### PR TITLE
API: Updates `/categories` api to list categories only and removes tags

### PR DESCRIPTION
**Pull request by pratap0007**
This patch includes following changes 
* Removes the association between Category and Tag table
  - Modifies category and tag table
  - Add Migration for tag table to drop category_id column
    and delete the category-tag constraints
  - Modifies category parsing logic 
  
* Updates `/categories` api to list categories only and removes tags

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/pkg/db/migration/migration.go [hunk 1]
```diff
@@ -36,6 +36,8 @@ func Migrate(api *app.APIBase) error {
 			addRefreshTokenChecksumColumnInUserTable(log),
 			updateCatalogBranchToMain(log),
 			addAvatarURLColumnInUsersTable(log),
+			removeCatgoryAndTagAssociation(log),
```
Comment (sm43) [inline api/pkg/db/migration/migration.go:39]: you can do both in single migration :)

File: api/pkg/service/category/category.go [hunk 2]
```diff
@@ -44,22 +43,15 @@ func (s *service) List(ctx context.Context) (*category.ListResult, error) {
...
 	var all []model.Category
-	if err := db.Order("name").
-		Preload("Tags", func(db *gorm.DB) *gorm.DB {
-			return db.Order("tags.name ASC")
-		}).
-		Find(&all).Error; err != nil {
+	if err := db.Model(&model.Category{}).Order("name").Find(&all).Error; err != nil {
 		log.Error(err)
 		return nil, fetchError
+
```
Comment (sm43) [inline api/pkg/service/category/category.go:49]: nit: empty line

**Comment by PuneetPunamiya** [comment]
Thanks 🤙🏻 

/lgtm

**Pull request merged**